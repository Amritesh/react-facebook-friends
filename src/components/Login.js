import React from 'react';
import Radium from 'radium';

const style = {
  btn: {
    backgroundColor: '#55acee',
    margin: '0 auto',
    color: 'white',
    width: 'inherit'
  },
  title: {
    color: '#55acee'
  },
  main: {
    backgroundColor: 'white',
    padding: '25px 5px',
    borderRadius: '3px'
  },
  wrapper: {
    textAlign: 'center',
    position: 'absolute',
    width: '40%',
    height: '20%',
    margin: 'auto',
    top: '20%',
    right: '30%',
    '@media (max-width: 48em)': {
      left: '0',
      width: '100%'
    }
  }
};

const Login = () => (
  <div style={style.wrapper}>
    <h2 style={style.title}>Checkout who are you best friends on Facebook</h2>
    <div style={style.main}>
      <h4>Facebook required your permission to do further action</h4>
      <button className="pure-button" style={style.btn} onClick={this.props.FB_login}>Agree</button>
    </div>
  </div>
);

export default Radium(Login);
