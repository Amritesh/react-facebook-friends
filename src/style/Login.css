.btn {
  background: #55acee;
  color: white;
}

.title {
  text-align: center;
  color: #55acee;
}

.main {
  text-align: center;
  background: white;
  padding: 15px 5px;
  border-radius: 3px;
}

.wrapper {
  display: flex;
  min-height: 60vh;
  align-items: center;
  justify-content: center;
}

@media (max-width: 600px) {
  .title {
    font-size: 1em;
  }

  .main {
    font-size: .9em;
  }
}