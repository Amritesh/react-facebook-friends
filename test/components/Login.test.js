import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../../src/components/Login';

test('Login component should renders correctly', () => {
  const props = {
    fBLogin: jest.fn()
  };
  const component = renderer.create(<Login {...props} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
