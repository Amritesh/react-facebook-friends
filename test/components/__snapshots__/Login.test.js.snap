// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should renders correctly 1`] = `
<div
  className="wrapper"
>
  <div>
    <h2
      className="title"
    >
      Checkout who are you best friends on Facebook
    </h2>
    <div
      className="main"
    >
      <h4>
        Facebook required your permission to do further action
      </h4>
      <button
        className="pure-button btn"
        onClick={[Function]}
      >
        Agree
      </button>
    </div>
  </div>
</div>
`;
